CraftTweaker2-API/src/main/java/crafttweaker/api/recipes/expand/ExpandCraftingRecipe.java

```diff
@@ -0,0 +1,19 @@
+package crafttweaker.api.recipes.expand;
+
+import crafttweaker.CraftTweakerAPI;
+import crafttweaker.annotations.ZenRegister;
+import crafttweaker.api.recipes.ICraftingRecipe;
+import stanhebben.zenscript.annotations.ZenExpansion;
+import stanhebben.zenscript.annotations.ZenMethod;
+
+/**
+ * @author youyihj
+ */
+@ZenRegister
+@ZenExpansion("crafttweaker.recipes.ICraftingRecipe")
+public class ExpandCraftingRecipe {
+    @ZenMethod
+    public static void remove(ICraftingRecipe recipe) {
+        CraftTweakerAPI.recipes.removeByRecipeName(recipe.getFullResourceName(), null);
```

Comment:
This won't work as recipes need to be staged in 1.12 to be removed correctly, also needs to be an IAction.

@kindlich thoughts on this method in general though?

Reply:
First, `removeByRecipeName` method will new an IAction to remove recipe. Second, we are only supposed to use the ICraftingRecipe class to tweak recipes on runnable scripts that loaded on game initialization, instead of in events and so on.

So the example script is that

```zenscript
for recipe in recipes.all {
    recipe.remove();
}
```

Does anyone try calling IRecipeManager on game avaliable? Like the script below:

```zenscript
events.onPlayerLoggedIn(function(event as PlayerLoggedInEvent) {
    recipes.remove(<item:minecraft:chest>);
});
``` 

The script won't work, too.

Reply:
I don't really see the need for this to be an explicit function in ICraftingRecipe?
Can't you just do
```zenscript
//imports ...;

function remove(recipe as ICraftingRecipe) as void {
    recipes.removeByRecipeName(recipe.fullResourceDomain);
}

for recipe in recipes.all {
    remove(recipe);
}
```

Reply:
It is fair